### ISSUE TITLE
Bundling JRE with Launch4j

### ISSUE THREAD

**Issue by entangledloops** [libgdx/libgdx#2799]
On [this](https://github.com/libgdx/libgdx/wiki/Bundling-a-jre) page you claim that Launch4j can be used to bundle a JRE with your JAR file. However, after spending about an hour using the tool and Googling around, I've discovered this to be far from simple. 

(For example, see [here](https://stackoverflow.com/questions/13996547/how-do-i-bundle-a-jre-into-an-exe-for-a-java-application-launch4j-says-runtime]), [here](https://stackoverflow.com/questions/15412530/how-do-i-use-the-bundle-jre-option-in-launch4j), and [here](https://stackoverflow.com/questions/6871987/distribute-my-application-jar-with-a-jre)).

The Launch4j tool lets you specify a relative "Bundled JRE Path", which sounds like a relative path within the jar file to the JRE. For example, within my jar I have included a folder "jre" which contains a complete JRE (v1.8.0_31). However, specifying "jre" as the path only tells Launch4j to look in the folder _outside_ the jar, instead of looking within the jar as you would expect from the word "bundled". Copious stackoverflow posts confirm that this is an issue, and that others have combined Launch4j along with another install wrapper so they can include the JRE folder adjacent their jar file after the install. This way, Launch4j can find the JRE. 

In short, this is neither desirable nor expected behavior: I want to include the entire JRE within my JAR, and point Launch4j into the jar file.

**TL;DR** **Please provide a brief tutorial or a few pointers describing how to bundle a JRE within a JAR and point Launch4j to that JRE.**

To confirm that you are actually using the desired JRE (not the system JRE and not the JRE in a local folder, but the actual JRE bundled within the jar), you should be able to view properties of the javaw process from Task Manager and see the path of the process is in some temporary extracted location, not in "Program Files\.....".

**Comment by badlogic** [comment]
Please give this a try: https://github.com/libgdx/packr

Feel free to patch up the Wiki page, it's editable by anyone.

**Comment by entangledloops** [comment]
Thanks for the quick response, will give it a try.
